src/main/java/com/blamejared/contenttweaker/api/blocks/IIsCoTBlock.java

```diff
@@ -1,6 +1,101 @@
 package com.blamejared.contenttweaker.api.blocks;
 
-import com.blamejared.contenttweaker.api.*;
-import net.minecraftforge.common.extensions.*;
+import com.blamejared.contenttweaker.actions.ActionSetFunction;
+import com.blamejared.contenttweaker.api.IHasCoTItem;
+import com.blamejared.contenttweaker.api.IHasResourceLocation;
+import com.blamejared.contenttweaker.api.IHasResourcesToWrite;
+import com.blamejared.contenttweaker.api.functions.*;
+import com.blamejared.contenttweaker.api.items.IIsCotItem;
+import com.blamejared.crafttweaker.api.annotations.ZenRegister;
+import com.blamejared.crafttweaker_annotations.annotations.Document;
+import net.minecraftforge.common.extensions.IForgeBlock;
+import org.openzen.zencode.java.ZenCodeType;
 
-public interface IIsCoTBlock extends IHasCoTItem, IHasResourcesToWrite, IHasResourceLocation, IForgeBlock {}
+/**
+ * A registered CoT Block. Used for advanced functionality. like onRandomTick, onReplaced etc.
+ * <p>
+ * These functions should be run in CraftTweaker scripts, instead of ContentTweaker ones. And they are reloadable.
+ * You can get it via cotItem BEP.
+ */
+@ZenRegister
+@Document("mods/contenttweaker/API/block/IIsCotBlock")
+@ZenCodeType.Name("mods.contenttweaker.block.IIsCotBlock")
+public interface IIsCoTBlock extends IHasCoTItem, IHasResourcesToWrite, IHasResourceLocation, IForgeBlock {
```

Comment:
Why extending IForgeBlock?

Reply:
https://github.com/CraftTweaker/ContentTweaker/blob/a6573c68a9a39203904686c108bd8e2294f30608/src/main/java/com/blamejared/contenttweaker/api/blocks/IIsCoTBlock.java#L6

it originally extends IForgeBlock. I don't change it.

Reply:
Odd, but alright.